fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/TreeLike.java

```diff
@@ -69,13 +70,27 @@ public interface TreeLike<T extends TreeLike<T>> {
     @Nonnull
     T withChildren(Iterable<? extends T> newChildren);
 
+    /**
+     * Returns an iterator that traverse the nodes in pre-order.
+     * @return an iterator that traverse the nodes in pre-order.
+     */
+    @Nonnull
+    default Iterator<T> preOrderIterator() {
+        return PreOrderIterator.over(getThis());
+    }
+
+    @Nonnull
+    default Stream<T> stream() {
```

Comment:
Please rename to `preOrderStream()` and then have all 50 callers use this instead of `Streams.stream(preOrderIterator())......`